sklearn/datasets/tests/test_openml.py

```diff
@@ -255,6 +258,248 @@ def _mock_urlopen(request):
         context.setattr(sklearn.datasets.openml, 'urlopen', _mock_urlopen)
 
 
+@pytest.mark.parametrize('feature, expected_dtype', [
+    ({'data_type': 'string', 'number_of_missing_values': '0'}, object),
+    ({'data_type': 'string', 'number_of_missing_values': '1'}, object),
+    ({'data_type': 'numeric', 'number_of_missing_values': '0'}, np.float64),
+    ({'data_type': 'numeric', 'number_of_missing_values': '1'}, np.float64),
+    ({'data_type': 'real', 'number_of_missing_values': '0'}, np.float64),
+    ({'data_type': 'real', 'number_of_missing_values': '1'}, np.float64),
+    ({'data_type': 'integer', 'number_of_missing_values': '0'}, np.int64),
+    ({'data_type': 'integer', 'number_of_missing_values': '1'}, np.float64),
+    ({'data_type': 'nominal', 'number_of_missing_values': '0'}, 'category'),
+    ({'data_type': 'nominal', 'number_of_missing_values': '1'}, 'category'),
+])
+def test_feature_to_dtype(feature, expected_dtype):
+    assert _feature_to_dtype(feature) == expected_dtype
+
+
+@pytest.mark.parametrize('feature', [
+    {'data_type': 'datatime', 'number_of_missing_values': '0'}
+])
+def test_feature_to_dtype_error(feature):
+    msg = 'Unsupported feature: {}'.format(feature)
+    with pytest.raises(ValueError, match=msg):
+        _feature_to_dtype(feature)
+
+
+def test_fetch_openml_iris_pandas(monkeypatch):
+    # classification dataset with numeric only columns
+    pd = pytest.importorskip('pandas')
+    data_id = 61
+    expected_shape = (150, 5)
+
+    cat_dtype = pd.CategoricalDtype(['Iris-setosa', 'Iris-versicolor',
+                                     'Iris-virginica'])
+    expected_dtypes = [np.float64] * 4 + [cat_dtype]
+    expected_feature_names = ['sepallength', 'sepalwidth', 'petallength',
+                              'petalwidth']
+    expected_target_names = ['class']
+    expected_columns = expected_feature_names + expected_target_names
+
+    _monkey_patch_webbased_functions(monkeypatch, data_id, True)
+
+    bunch = fetch_openml(data_id=data_id, return_frame=True, cache=False)
```

Comment:
I think it would be best to compare the bunches returned with return_frame=False vs True. This is especially needed in case we were to add something to the bunch later...